robotbase/src/main/java/frc/robot/commands/pivot/PivotZero.java

```diff
@@ -1,33 +1,43 @@
 package frc.robot.commands.pivot;
 
-import edu.wpi.first.wpilibj.DriverStation;
 import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.ParallelDeadlineGroup;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
 import frc.robot.Constants;
 import frc.robot.Robot;
+import frc.robot.controls.util.AxisInterface;
 
-public class PivotZero extends Command {
-  public boolean isZeroed() {
-    return Robot.pivot.getPivotVelocity() <= Constants.SHOOTER.ZERO_SPEED_STOP_TOLERANCE;
-  }
-
-  @Override
-  public void initialize() {
-    Robot.pivot.setPivotAxisSpeed(Constants.SHOOTER.ZERO_SPEED);
-  }
-
-  @Override
-  public boolean isFinished() {
-    return isZeroed();
+public class PivotZero extends ParallelDeadlineGroup {
+  public static class UntilPivotZero extends Command {
+    @Override
+    public boolean isFinished() {
+      return Math.abs(Robot.pivot.getVelocity()) < Constants.PIVOT.ZERO_SPEED_STOP_TOLERANCE;
+    }
   }
 
-  @Override
-  public void end(boolean interrupted) {
-    Robot.pivot.stop();
+  public PivotZero() {
+    super(
+        // First wait until we start moving, then wait until the velocity stops
+        new SequentialCommandGroup(
+                new WaitCommand(Constants.PIVOT.ZERO_SPEED_INITIAL_SECONDS), new UntilPivotZero())
+            .finallyDo(
+                (boolean interrupted) -> {
+                  if (!interrupted) {
+                    Robot.pivot.setZero();
+                  } else {
+                    System.err.println("Pivot Zero interrupted!");
+                  }
+                }),
 
-    if (!interrupted) {
-      Robot.pivot.zero();
-    } else {
-      DriverStation.reportError("Pivot Zero interrupted!", false);
-    }
+        // Pivot along the axis until the above finishes
+        new PivotAxis(
+            new AxisInterface() {
```

Comment:
I think we should use a seperate command that is just `PivotPercentOut` or `PivotSpeed` that sets percent out on the motor. The Pivot.setAxisSpeed() method the `PivotAxis` command calls is intended for input from a stick. The `PivotAxis` command also calls `Pivot.setAxisSpeed()` every execute which is not necessary here. It inverts the input and constrains it against a AXIS_MAX_SPEED. I think it would be clearer to just use percent output here without any additional modification. We could also lift what we are doing inside Pivot.setAxisSpeed() to the controller level.

Reply:
That's fair. I was on the fence myself when I wrote it. I'll adjust the subsystem to not deal in axis units and put the axis unit conversion in a command.